rosidl_generator_c/src/string_functions.c

```diff
@@ -43,7 +44,7 @@ rosidl_generator_c__String__fini(rosidl_generator_c__String * str)
     str->size = 0;
     str->capacity = 0;
   } else {
-    /* ensure that data, szie and and capacity values are consistent */
+    /* ensure that data, size, and capacity values are consistent */
     assert(0 == str->size);
     assert(0 == str->capacity);
```

Comment:
should the asserts on the capacity use 1 instead of 0 for string checking?
e.g. line 40/41 `assert(str->capacity > 1);` and line 48/49 `assert(1 == str->capacity);`